### PR TITLE
Use an `Options` struct to pass options around, instead of needing to pass kwargs around everywhere

### PR DESCRIPTION
**Pull request by DilumAluthge**
Instead of needing to pass kwargs around everywhere, let's put all the kwargs into a single `Options` struct, and then we can just pass a single `options::Options` around, instead of needing to pass all of the kwargs around.

### CODE REVIEW

File: src/dependencies.jl [hunk 1]
```diff
@@ -4,23 +4,22 @@ function get_project_deps(
     api::Forge,
     ci::CIService,
     repo::Union{GitHub.Repo,GitLab.Project};
-    subdir::AbstractString="",
-    include_jll::Bool=false,
-    master_branch::Union{DefaultBranch,AbstractString}=DefaultBranch(),
+    options::Options,
+    subdir::String,
```
Comment (mattBrzezinski) [inline src/dependencies.jl:8]:
```suggestion
    subdir::AbstractString,
```

File: src/main.jl [hunk 2]
```diff
@@ -45,54 +39,45 @@ Main entry point for the package.
...
+)
+    options = Options(; kwargs...)
+    return _true_main(env, ci_cfg; options)
+
+end
+
+function _true_main(
+    env::AbstractDict,
+    ci_cfg::CIService;
+    options::Options,
```
Comment (mattBrzezinski) [inline src/main.jl:52]:
```suggestion
    kwargs...,
)
    options = Options(; kwargs...)
```

File: src/main.jl [hunk 3]
```diff
@@ -45,54 +39,45 @@ Main entry point for the package.
...
-            subdir=subdir,
-            include_jll=include_jll,
-            master_branch=master_branch,
+            options,
+            subdir,
         )
 
-        if use_existing_registries
+        if options.use_existing_registries
             get_existing_registries!(deps, depot)
```
Comment (mattBrzezinski) [inline src/main.jl:68]:
```suggestion
            get_existing_registries!(deps, options.depot)
```

**Review by mattBrzezinski — COMMENTED** [review]
Might need to pull this locally down to review, GH makes it a bit tricky to review

**Comment by DilumAluthge** [comment]
bors try

**Comment by DilumAluthge** [comment]
@mattBrzezinski @fchorney I'm going to merge this now, just because I want to get #418 in soon, but feel free to continue reviewing post-merge; we can apply any necessary changes in follow-up PRs.

**Comment by DilumAluthge** [comment]
bors merge

**Review by fchorney — APPROVED** [review]
Overall I like the idea, passing around all those options was definitely annoying. 

I'm a little unsure why we're using a pipe `|>` for `options.subdirs |> only` instead of just using `only(options.subdirs)` is there something that the pipe gives us that using parens doesn't?

**Comment by DilumAluthge** [comment]
> I'm a little unsure why we're using a pipe `|>` for `options.subdirs |> only` instead of just using `only(options.subdirs)` is there something that the pipe gives us that using parens doesn't?

I just thought I'd try something new 😂 

If you found it confusing, that's probably a sign that we should switch back to the `only(options.subdirs)` syntax.

**Pull request merged**